### PR TITLE
feat(plugin/packetparser): identify CPU as Flow metadata

### LINKED ISSUE #94: Dropped flows should identify CPU
**Is your feature request related to a problem? Please describe.**

Recently, we saw an issue where inside a particular node, packets were dropped on a particular CPU. However, kubectl top was not showing CPU running hot (because only few CPUs out of 32 were over burdened). Add the CPU number to the flow extension for drop.

cc: @anubhabMajumdar

### PR DESCRIPTION
**Pull request by hainenber**
# Description

Add CPU number as Flow metadata

## Related Issue

Closes #94 

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [ ] I have tested the changes locally.
- [ ] I have followed the project's style guidelines.
- [ ] I have updated the documentation, if necessary.
- [x] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

File: pkg/plugin/packetparser/packetparser_linux.go [hunk 1]
```diff
@@ -553,6 +553,9 @@ func (p *packetParser) processRecord(ctx context.Context, id int) {
 			tcpMetadata := bpfEvent.TcpMetadata
 			utils.AddTcpFlags(fl, tcpMetadata.Syn, tcpMetadata.Ack, tcpMetadata.Fin, tcpMetadata.Rst, tcpMetadata.Psh, tcpMetadata.Urg)
 
+			// Add CPU metadata to the flow.
+			utils.AddCpu(fl, uint32(record.CPU))
```
Comment (anubhabMajumdar) [inline pkg/plugin/packetparser/packetparser_linux.go:557]: We also need to add this for `DropReason` plugin.

**Review by anubhabMajumdar — COMMENTED** [review]
Thanks for the PR. Can you add `Testing Done` section, especially testing around verifying the CPU variable is correct?

File: pkg/plugin/packetparser/packetparser_linux.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hainenber) [reply pkg/plugin/packetparser/packetparser_linux.go:557]: I saw that there are existing `utils.Add*` functions, each having its own `Marshal` call.  Do you think we should refactor them also in this PR?

**Comment by hainenber** [comment]
> Thanks for the PR. Can you add Testing Done section, especially testing around verifying the CPU variable is correct?

Sure, I'll try get the cluster up and running for a proper E2E test. I was thinking that unit test would be sufficient 😅

File: pkg/plugin/packetparser/packetparser_linux.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (anubhabMajumdar) [reply pkg/plugin/packetparser/packetparser_linux.go:557]: Just `AddCpu` for now, you can file an issue for further refactor and handle it separately?

**Pull request closed without merging**